transport/src/main/java/com/strategyobject/substrateclient/transport/ws/ReconnectionPolicy.java

```diff
@@ -0,0 +1,43 @@
+package com.strategyobject.substrateclient.transport.ws;
+
+import lombok.NonNull;
+
+/**
+ * Represents a strategy of reconnection
+ */
+public interface ReconnectionPolicy {
+
+    /**
+     * The method is called when connection was closed and probably should be reconnected
+     * @param context contains a reason of disconnection
+     * @return a schedule planned for reconnection
+     */
+    @NonNull ReconnectionSchedule apply(@NonNull ReconnectionContext context);
+
+    /**
+     * The method is called when connection successfully open
+     */
+    void reset();
```

Comment:
I think this method implies that the policy is stateful. I can suggest doing it a little different. The policy holds an algorithm, which generates a schedule providing you with next timeslots. If you need to reset you just generate a new schedule.

```
public interface ReconnectionPolicy {
    interface ReconnectionSchedule {
        long getNextDelay();
        TimeUnit getTimeUnit();
    }

    ReconnectionSchedule getSchedule();
}
```

Reply:
Also, it probably might be better if a consumer can tell the schedule which time units it wants to see the results in instead of being made to use time units predefined by the author of the policy.
`long getNextDelay(TimeUnit timeUnit)`